cmd/build.go

```diff
@@ -173,6 +175,45 @@ func GenerateKustomizeResult(config opConfig.Config, kustomizeTemplate template.
 		yamlFile.PutWithSeparator("metalLbSecretKey", base64.StdEncoding.EncodeToString(metalLbSecretKey), ".")
 	}
 
+	artifactRepoS3Node, _ := yamlFile.Get("artifactRepository.s3")
```

Comment:
Haven't fully tested this, but can this section be refactored like so?
```
        _, artifactRepositoryNode := yamlFile.Get("artifactRepository")
	artifactRepositoryConfig := v1.ArtifactRepositoryConfig{}
	artifactRepositoryNode.Decode(artifactRepositoryConfig)
	if err != nil {
		return "", err
	}
	if artifactRepositoryConfig.S3 != nil {
		artifactRepositoryConfig.S3.AccessKeySecret.Key = artifactRepositoryConfig.S3.AccessKey
		artifactRepositoryConfig.S3.AccessKeySecret.Name = "$(artifactRepositoryS3AccessKeySecretName)"
		artifactRepositoryConfig.S3.SecretKeySecret.Key = artifactRepositoryConfig.S3.Secretkey
		artifactRepositoryConfig.S3.SecretKeySecret.Name = "$(artifactRepositoryS3SecretKeySecretName)"
		err, yamlStr := artifactRepositoryConfig.S3.MarshalToYaml()
		if err != nil {
			return "", err
		}
		yamlFile.Put("artifactRepositoryProvider", yamlStr)
	}
	else if artifactRepositoryConfig.GCS != nil {
		err, yamlConfigMap := artifactRepositoryConfig.GCS.MarshalToYaml()
		if err != nil {
			return "", err
		}

		yamlFile.Put("artifactRepositoryProvider", yamlConfigMap)
	} else {
		return "", errors.New("unsupported artifactRepository configuration")
	}
```

Reply:
Also, it looks like GCS is still not using secrets?

Reply:
This is implemented.